client/web/src/repo/blob/own/FileOwnershipEntry.tsx

```diff
@@ -23,7 +23,17 @@ export const FileOwnershipEntry: React.FunctionComponent<Props> = ({ owner, reas
         setIsExpanded(!isExpanded)
     }, [isExpanded])
 
-    const email = owner.__typename === 'Person' ? owner.email : undefined
+    const findEmail = (): string | undefined => {
+        if (owner.__typename !== 'Person') {
+            return undefined
+        }
+        if (owner.user?.primaryEmail !== null) {
+            return owner.user?.primaryEmail.email
+        }
+        return owner.email
+    }
+
+    const email = findEmail()
```

Comment:
nit: maybe we should memoize this?

```suggestion
    const email = useMemo(() => {
        if (owner.__typename !== 'Person') {
            return undefined
        }
        if (owner.user?.primaryEmail !== null) {
            return owner.user?.primaryEmail.email
        }
        return owner.email
    }, [owner])
```

Reply:
It's not computational heavy, so we trade 2 conditionals vs 1 equality check on an array for the deps change check, which is a deep equals, sort of. Do you think that's worth it?

Reply:
I'd think it's more matter of style than performance. I'm happy to go either way. To my lay eye the original way looks a little scrappy, and with the memo looks more react :)

Reply:
I think either way it's fine, `useMemo` has some extra overhead so not sure if it's a perf improvement.